gpii/node_modules/flowManager/src/Capture.js

```diff
@@ -0,0 +1,283 @@
+/**
+ * GPII snapshot Handler
+ *
+ * Copyright 2014 Raising the Floor - International
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/gpii/universal/LICENSE.txt
+ */
+"use strict";
+
+var fluid = require("infusion"),
+    gpii = fluid.registerNamespace("gpii");
+
+fluid.defaults("gpii.flowManager.capture", {
+    gradeNames: ["fluid.component"],
+    events: {
+        onPrepareCapture: null,
+        onPrepareSolutions: null,
+        onSolutions: null,
+        onReadyToCapture: {
+            events: {
+                prepareCapture: "onPrepareCapture",
+                prepareSolutions: "onPrepareSolutions",
+                solutions: "onSolutions"
+            },
+            args: [
+                "{arguments}.solutions.0",
+                "{arguments}.prepareCapture.0"
+            ]
+        },
+        onError: null,
+
+        // Newer versions below using fireTransformEvent
+        onSolutionsForCurrentDevice: null,
+        onCaptureSettingsForCurrentDevice: null
+    },
+    listeners: {
+        onSolutionsForCurrentDevice: [
+            {
+                funcName: "gpii.lifecycleManager.getDeviceContextPromise",
+                args: ["{flowManager}.deviceReporter"]
+            },
+            {
+                funcName: "gpii.flowManager.getSolutionsPromise",
+                args: [ "{flowManager}.solutionsRegistryDataSource", "{arguments}.0"]
+            },
+            {
+                funcName: "fluid.toPromise",
+                args: ["{arguments}.0.solutionsRegistryEntries"]
+            }
+        ],
+        onCaptureSettingsForCurrentDevice: [
+            {
```

Comment:
Line 56-67 is a replica of line 42-53 and can be replaced by calling `gpii.flowManager.getInstalledSolutions()` if `getInstalledSolutions()` is an API you need in the capture tool in gpii-app. The search in the code base shows `getInstalledSolutions()` is not used in the universal. If you don't need it elsewhere, `getInstalledSolutions()` along with `onSolutionsForCurrentDevice` event can be removed.

Also, please add tests for `getInstalledSolutions()` if you decide to keep it.

Reply:
I've removed the replica, and added a test for getInstalledSolutions, as it is an API needed by the capture tool outside universal.